patroni/postgresql/config.py

```diff
@@ -618,19 +618,19 @@ def _read_recovery_params_pre_v12(self):
 
     def _check_passfile(self, passfile, wanted_primary_conninfo):
         # If there is a passfile in the primary_conninfo try to figure out that
-        # the passfile contains the line allowing connection to the given node.
+        # the passfile contains the line(s) allowing connection to the given node.
         # We assume that the passfile was created by Patroni and therefore doing
         # the full match and not covering cases when host, port or user are set to '*'
         passfile_mtime = mtime(passfile)
         if passfile_mtime:
             try:
                 with open(passfile) as f:
-                    wanted_line = self._pgpass_line(wanted_primary_conninfo).strip()
-                    for raw_line in f:
-                        if raw_line.strip() == wanted_line:
-                            self._passfile = passfile
-                            self._passfile_mtime = passfile_mtime
-                            return True
+                    wanted_lines = self._pgpass_line(wanted_primary_conninfo).splitlines()
+                    file_lines = f.read().splitlines()
+                    if (wanted_lines == file_lines):
```

Comment:
Since the order of lines doesn't really matter in the pgpassfile we better compare sets:
```suggestion
                    if set(wanted_lines) == set(file_lines):
```

Reply:
Thanks, did it this way in [3c7aac7](https://github.com/zalando/patroni/pull/2192/commits/3c7aac785a04d36dde7f2b673b2bb434b5e8b53c)